File: src/libqsize.cpp
#include <QMargins>
#include <QMarginsF>
#include <QSize>
#include <QSizeF>
#include <qsize.h>
#include "libqsize.h"
#include "libqsize.hxx"

QSize* QSize_new(QSize* other) {
    return new QSize(*other);
}

QSize* QSize_new2(QSize* other) {
    return new QSize(std::move(*other));
}

QSize* QSize_new3() {
    return new QSize();
}

QSize* QSize_new4(int w, int h) {
    return new QSize(static_cast<int>(w), static_cast<int>(h));
}

QSize* QSize_new5(QSize* param1) {
    return new QSize(*param1);
}

void QSize_CopyAssign(QSize* self, QSize* other) {
    *self = *other;
}

void QSize_MoveAssign(QSize* self, QSize* other) {
    *self = std::move(*other);
}

bool QSize_IsNull(const QSize* self) {
    return self->isNull();
}

bool QSize_IsEmpty(const QSize* self) {
    return self->isEmpty();
}

bool QSize_IsValid(const QSize* self) {
    return self->isValid();
}

int QSize_Width(const QSize* self) {
    return self->width();
}

int QSize_Height(const QSize* self) {
    return self->height();
}

void QSize_SetWidth(QSize* self, int w) {
    self->setWidth(static_cast<int>(w));
}

void QSize_SetHeight(QSize* self, int h) {
    self->setHeight(static_cast<int>(h));
}

void QSize_Transpose(QSize* self) {
    self->transpose();
}

QSize* QSize_Transposed(const QSize* self) {
    return new QSize(self->transposed());
}

void QSize_Scale(QSize* self, int w, int h, int mode) {
    self->scale(static_cast<int>(w), static_cast<int>(h), static_cast<Qt::AspectRatioMode>(mode));
}

void QSize_Scale2(QSize* self, QSize* s, int mode) {
    self->scale(*s, static_cast<Qt::AspectRatioMode>(mode));
}

QSize* QSize_Scaled(const QSize* self, int w, int h, int mode) {
    return new QSize(self->scaled(static_cast<int>(w), static_cast<int>(h), static_cast<Qt::AspectRatioMode>(mode)));
}

QSize* QSize_Scaled2(const QSize* self, QSize* s, int mode) {
    return new QSize(self->scaled(*s, static_cast<Qt::AspectRatioMode>(mode)));
}

QSize* QSize_ExpandedTo(const QSize* self, QSize* param1) {
    return new QSize(self->expandedTo(*param1));
}

QSize* QSize_BoundedTo(const QSize* self, QSize* param1) {
    return new QSize(self->boundedTo(*param1));
}

QSize* QSize_GrownBy(const QSize* self, QMargins* m) {
    return new QSize(self->grownBy(*m));
}

QSize* QSize_ShrunkBy(const QSize* self, QMargins* m) {
    return new QSize(self->shrunkBy(*m));
}

QSize* QSize_OperatorPlusAssign(QSize* self, QSize* param1) {
    QSize& _ret = self->operator+=(*param1);
    // Cast returned reference into pointer
    return &_ret;
}

QSize* QSize_OperatorMinusAssign(QSize* self, QSize* param1) {
    QSize& _ret = self->operator-=(*param1);
    // Cast returned reference into pointer
    return &_ret;
}

QSize* QSize_OperatorMultiplyAssign(QSize* self, double c) {
    QSize& _ret = self->operator*=(static_cast<qreal>(c));
    // Cast returned reference into pointer
    return &_ret;
}

QSize* QSize_OperatorDivideAssign(QSize* self, double c) {
    QSize& _ret = self->operator/=(static_cast<qreal>(c));
    // Cast returned reference into pointer
    return &_ret;
}

QSizeF* QSize_ToSizeF(const QSize* self) {
    return new QSizeF(self->toSizeF());
}

void QSize_Delete(QSize* self) {
    delete self;
}

QSizeF* QSizeF_new(QSizeF* other) {
    return new QSizeF(*other);
}

QSizeF* QSizeF_new2(QSizeF* other) {
    return new QSizeF(std::move(*other));
}

QSizeF* QSizeF_new3() {
    return new QSizeF();
}

QSizeF* QSizeF_new4(QSize* sz) {
    return new QSizeF(*sz);
}

QSizeF* QSizeF_new5(double w, double h) {
    return new QSizeF(static_cast<qreal>(w), static_cast<qreal>(h));
}

QSizeF* QSizeF_new6(QSizeF* param1) {
    return new QSizeF(*param1);
}

void QSizeF_CopyAssign(QSizeF* self, QSizeF* other) {
    *self = *other;
}

void QSizeF_MoveAssign(QSizeF* self, QSizeF* other) {
    *self = std::move(*other);
}

bool QSizeF_IsNull(const QSizeF* self) {
    return self->isNull();
}

bool QSizeF_IsEmpty(const QSizeF* self) {
    return self->isEmpty();
}

bool QSizeF_IsValid(const QSizeF* self) {
    return self->isValid();
}

double QSizeF_Width(const QSizeF* self) {
    return static_cast<double>(self->width());
}

double QSizeF_Height(const QSizeF* self) {
    return static_cast<double>(self->height());
}

void QSizeF_SetWidth(QSizeF* self, double w) {
    self->setWidth(static_cast<qreal>(w));
}

void QSizeF_SetHeight(QSizeF* self, double h) {
    self->setHeight(static_cast<qreal>(h));
}

void QSizeF_Transpose(QSizeF* self) {
    self->transpose();
}

QSizeF* QSizeF_Transposed(const QSizeF* self) {
    return new QSizeF(self->transposed());
}

void QSizeF_Scale(QSizeF* self, double w, double h, int mode) {
    self->scale(static_cast<qreal>(w), static_cast<qreal>(h), static_cast<Qt::AspectRatioMode>(mode));
}

void QSizeF_Scale2(QSizeF* self, QSizeF* s, int mode) {
    self->scale(*s, static_cast<Qt::AspectRatioMode>(mode));
}

QSizeF* QSizeF_Scaled(const QSizeF* self, double w, double h, int mode) {
    return new QSizeF(self->scaled(static_cast<qreal>(w), static_cast<qreal>(h), static_cast<Qt::AspectRatioMode>(mode)));
}

QSizeF* QSizeF_Scaled2(const QSizeF* self, QSizeF* s, int mode) {
    return new QSizeF(self->scaled(*s, static_cast<Qt::AspectRatioMode>(mode)));
}

QSizeF* QSizeF_ExpandedTo(const QSizeF* self, QSizeF* param1) {
    return new QSizeF(self->expandedTo(*param1));
}

QSizeF* QSizeF_BoundedTo(const QSizeF* self, QSizeF* param1) {
    return new QSizeF(self->boundedTo(*param1));
}

QSizeF* QSizeF_GrownBy(const QSizeF* self, QMarginsF* m) {
    return new QSizeF(self->grownBy(*m));
}

QSizeF* QSizeF_ShrunkBy(const QSizeF* self, QMarginsF* m) {
    return new QSizeF(self->shrunkBy(*m));
}

QSizeF* QSizeF_OperatorPlusAssign(QSizeF* self, QSizeF* param1) {
    QSizeF& _ret = self->operator+=(*param1);
    // Cast returned reference into pointer
    return &_ret;
}

QSizeF* QSizeF_OperatorMinusAssign(QSizeF* self, QSizeF* param1) {
    QSizeF& _ret = self->operator-=(*param1);
    // Cast returned reference into pointer
    return &_ret;
}

QSizeF* QSizeF_OperatorMultiplyAssign(QSizeF* self, double c) {
    QSizeF& _ret = self->operator*=(static_cast<qreal>(c));
    // Cast returned reference into pointer
    return &_ret;
}

QSizeF* QSizeF_OperatorDivideAssign(QSizeF* self, double c) {
    QSizeF& _ret = self->operator/=(static_cast<qreal>(c));
    // Cast returned reference into pointer
    return &_ret;
}

QSize* QSizeF_ToSize(const QSizeF* self) {
    return new QSize(self->toSize());
}

void QSizeF_Delete(QSizeF* self) {
    delete self;
}

File: src/libqsize.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const qnamespace_enums = @import("libqnamespace.zig").enums;

/// https://doc.qt.io/qt-6/qsize.html
pub const qsize = struct {
    /// New constructs a new QSize object.
    ///
    /// ``` other: QtC.QSize ```
    pub fn New(other: ?*anyopaque) QtC.QSize {
        return qtc.QSize_new(@ptrCast(other));
    }

    /// New2 constructs a new QSize object and invalidates the source QSize object.
    ///
    /// ``` other: QtC.QSize ```
    pub fn New2(other: ?*anyopaque) QtC.QSize {
        return qtc.QSize_new2(@ptrCast(other));
    }

    /// New3 constructs a new QSize object.
    ///
    ///
    pub fn New3() QtC.QSize {
        return qtc.QSize_new3();
    }

    /// New4 constructs a new QSize object.
    ///
    /// ``` w: i32, h: i32 ```
    pub fn New4(w: i32, h: i32) QtC.QSize {
        return qtc.QSize_new4(@intCast(w), @intCast(h));
    }

    /// New5 constructs a new QSize object.
    ///
    /// ``` param1: QtC.QSize ```
    pub fn New5(param1: ?*anyopaque) QtC.QSize {
        return qtc.QSize_new5(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    ///
    /// ``` self: QtC.QSize, other: QtC.QSize ```
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSize_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    ///
    /// ``` self: QtC.QSize, other: QtC.QSize ```
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSize_MoveAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#isNull)
    ///
    /// ``` self: QtC.QSize ```
    pub fn IsNull(self: ?*anyopaque) bool {
        return qtc.QSize_IsNull(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#isEmpty)
    ///
    /// ``` self: QtC.QSize ```
    pub fn IsEmpty(self: ?*anyopaque) bool {
        return qtc.QSize_IsEmpty(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#isValid)
    ///
    /// ``` self: QtC.QSize ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.QSize_IsValid(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#width)
    ///
    /// ``` self: QtC.QSize ```
    pub fn Width(self: ?*anyopaque) i32 {
        return qtc.QSize_Width(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#height)
    ///
    /// ``` self: QtC.QSize ```
    pub fn Height(self: ?*anyopaque) i32 {
        return qtc.QSize_Height(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#setWidth)
    ///
    /// ``` self: QtC.QSize, w: i32 ```
    pub fn SetWidth(self: ?*anyopaque, w: i32) void {
        qtc.QSize_SetWidth(@ptrCast(self), @intCast(w));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#setHeight)
    ///
    /// ``` self: QtC.QSize, h: i32 ```
    pub fn SetHeight(self: ?*anyopaque, h: i32) void {
        qtc.QSize_SetHeight(@ptrCast(self), @intCast(h));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#transpose)
    ///
    /// ``` self: QtC.QSize ```
    pub fn Transpose(self: ?*anyopaque) void {
        qtc.QSize_Transpose(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#transposed)
    ///
    /// ``` self: QtC.QSize ```
    pub fn Transposed(self: ?*anyopaque) QtC.QSize {
        return qtc.QSize_Transposed(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#scale)
    ///
    /// ``` self: QtC.QSize, w: i32, h: i32, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scale(self: ?*anyopaque, w: i32, h: i32, mode: i64) void {
        qtc.QSize_Scale(@ptrCast(self), @intCast(w), @intCast(h), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#scale)
    ///
    /// ``` self: QtC.QSize, s: QtC.QSize, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scale2(self: ?*anyopaque, s: ?*anyopaque, mode: i64) void {
        qtc.QSize_Scale2(@ptrCast(self), @ptrCast(s), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#scaled)
    ///
    /// ``` self: QtC.QSize, w: i32, h: i32, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scaled(self: ?*anyopaque, w: i32, h: i32, mode: i64) QtC.QSize {
        return qtc.QSize_Scaled(@ptrCast(self), @intCast(w), @intCast(h), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#scaled)
    ///
    /// ``` self: QtC.QSize, s: QtC.QSize, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scaled2(self: ?*anyopaque, s: ?*anyopaque, mode: i64) QtC.QSize {
        return qtc.QSize_Scaled2(@ptrCast(self), @ptrCast(s), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#expandedTo)
    ///
    /// ``` self: QtC.QSize, param1: QtC.QSize ```
    pub fn ExpandedTo(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSize {
        return qtc.QSize_ExpandedTo(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#boundedTo)
    ///
    /// ``` self: QtC.QSize, param1: QtC.QSize ```
    pub fn BoundedTo(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSize {
        return qtc.QSize_BoundedTo(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#grownBy)
    ///
    /// ``` self: QtC.QSize, m: QtC.QMargins ```
    pub fn GrownBy(self: ?*anyopaque, m: QtC.QMargins) QtC.QSize {
        return qtc.QSize_GrownBy(@ptrCast(self), @ptrCast(m));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#shrunkBy)
    ///
    /// ``` self: QtC.QSize, m: QtC.QMargins ```
    pub fn ShrunkBy(self: ?*anyopaque, m: QtC.QMargins) QtC.QSize {
        return qtc.QSize_ShrunkBy(@ptrCast(self), @ptrCast(m));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#operator+=)
    ///
    /// ``` self: QtC.QSize, param1: QtC.QSize ```
    pub fn OperatorPlusAssign(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSize {
        return qtc.QSize_OperatorPlusAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#operator-=)
    ///
    /// ``` self: QtC.QSize, param1: QtC.QSize ```
    pub fn OperatorMinusAssign(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSize {
        return qtc.QSize_OperatorMinusAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#operator*=)
    ///
    /// ``` self: QtC.QSize, c: f64 ```
    pub fn OperatorMultiplyAssign(self: ?*anyopaque, c: f64) QtC.QSize {
        return qtc.QSize_OperatorMultiplyAssign(@ptrCast(self), @floatCast(c));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#operator/=)
    ///
    /// ``` self: QtC.QSize, c: f64 ```
    pub fn OperatorDivideAssign(self: ?*anyopaque, c: f64) QtC.QSize {
        return qtc.QSize_OperatorDivideAssign(@ptrCast(self), @floatCast(c));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#toSizeF)
    ///
    /// ``` self: QtC.QSize ```
    pub fn ToSizeF(self: ?*anyopaque) QtC.QSizeF {
        return qtc.QSize_ToSizeF(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsize.html#dtor.QSize)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QSize ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSize_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qsizef.html
pub const qsizef = struct {
    /// New constructs a new QSizeF object.
    ///
    /// ``` other: QtC.QSizeF ```
    pub fn New(other: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_new(@ptrCast(other));
    }

    /// New2 constructs a new QSizeF object and invalidates the source QSizeF object.
    ///
    /// ``` other: QtC.QSizeF ```
    pub fn New2(other: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_new2(@ptrCast(other));
    }

    /// New3 constructs a new QSizeF object.
    ///
    ///
    pub fn New3() QtC.QSizeF {
        return qtc.QSizeF_new3();
    }

    /// New4 constructs a new QSizeF object.
    ///
    /// ``` sz: QtC.QSize ```
    pub fn New4(sz: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_new4(@ptrCast(sz));
    }

    /// New5 constructs a new QSizeF object.
    ///
    /// ``` w: f64, h: f64 ```
    pub fn New5(w: f64, h: f64) QtC.QSizeF {
        return qtc.QSizeF_new5(@floatCast(w), @floatCast(h));
    }

    /// New6 constructs a new QSizeF object.
    ///
    /// ``` param1: QtC.QSizeF ```
    pub fn New6(param1: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_new6(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    ///
    /// ``` self: QtC.QSizeF, other: QtC.QSizeF ```
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSizeF_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    ///
    /// ``` self: QtC.QSizeF, other: QtC.QSizeF ```
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSizeF_MoveAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#isNull)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn IsNull(self: ?*anyopaque) bool {
        return qtc.QSizeF_IsNull(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#isEmpty)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn IsEmpty(self: ?*anyopaque) bool {
        return qtc.QSizeF_IsEmpty(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#isValid)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.QSizeF_IsValid(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#width)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn Width(self: ?*anyopaque) f64 {
        return qtc.QSizeF_Width(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#height)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn Height(self: ?*anyopaque) f64 {
        return qtc.QSizeF_Height(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#setWidth)
    ///
    /// ``` self: QtC.QSizeF, w: f64 ```
    pub fn SetWidth(self: ?*anyopaque, w: f64) void {
        qtc.QSizeF_SetWidth(@ptrCast(self), @floatCast(w));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#setHeight)
    ///
    /// ``` self: QtC.QSizeF, h: f64 ```
    pub fn SetHeight(self: ?*anyopaque, h: f64) void {
        qtc.QSizeF_SetHeight(@ptrCast(self), @floatCast(h));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#transpose)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn Transpose(self: ?*anyopaque) void {
        qtc.QSizeF_Transpose(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#transposed)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn Transposed(self: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_Transposed(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#scale)
    ///
    /// ``` self: QtC.QSizeF, w: f64, h: f64, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scale(self: ?*anyopaque, w: f64, h: f64, mode: i64) void {
        qtc.QSizeF_Scale(@ptrCast(self), @floatCast(w), @floatCast(h), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#scale)
    ///
    /// ``` self: QtC.QSizeF, s: QtC.QSizeF, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scale2(self: ?*anyopaque, s: ?*anyopaque, mode: i64) void {
        qtc.QSizeF_Scale2(@ptrCast(self), @ptrCast(s), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#scaled)
    ///
    /// ``` self: QtC.QSizeF, w: f64, h: f64, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scaled(self: ?*anyopaque, w: f64, h: f64, mode: i64) QtC.QSizeF {
        return qtc.QSizeF_Scaled(@ptrCast(self), @floatCast(w), @floatCast(h), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#scaled)
    ///
    /// ``` self: QtC.QSizeF, s: QtC.QSizeF, mode: qnamespace_enums.AspectRatioMode ```
    pub fn Scaled2(self: ?*anyopaque, s: ?*anyopaque, mode: i64) QtC.QSizeF {
        return qtc.QSizeF_Scaled2(@ptrCast(self), @ptrCast(s), @intCast(mode));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#expandedTo)
    ///
    /// ``` self: QtC.QSizeF, param1: QtC.QSizeF ```
    pub fn ExpandedTo(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_ExpandedTo(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#boundedTo)
    ///
    /// ``` self: QtC.QSizeF, param1: QtC.QSizeF ```
    pub fn BoundedTo(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_BoundedTo(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#grownBy)
    ///
    /// ``` self: QtC.QSizeF, m: QtC.QMarginsF ```
    pub fn GrownBy(self: ?*anyopaque, m: QtC.QMarginsF) QtC.QSizeF {
        return qtc.QSizeF_GrownBy(@ptrCast(self), @ptrCast(m));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#shrunkBy)
    ///
    /// ``` self: QtC.QSizeF, m: QtC.QMarginsF ```
    pub fn ShrunkBy(self: ?*anyopaque, m: QtC.QMarginsF) QtC.QSizeF {
        return qtc.QSizeF_ShrunkBy(@ptrCast(self), @ptrCast(m));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#operator+=)
    ///
    /// ``` self: QtC.QSizeF, param1: QtC.QSizeF ```
    pub fn OperatorPlusAssign(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_OperatorPlusAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#operator-=)
    ///
    /// ``` self: QtC.QSizeF, param1: QtC.QSizeF ```
    pub fn OperatorMinusAssign(self: ?*anyopaque, param1: ?*anyopaque) QtC.QSizeF {
        return qtc.QSizeF_OperatorMinusAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#operator*=)
    ///
    /// ``` self: QtC.QSizeF, c: f64 ```
    pub fn OperatorMultiplyAssign(self: ?*anyopaque, c: f64) QtC.QSizeF {
        return qtc.QSizeF_OperatorMultiplyAssign(@ptrCast(self), @floatCast(c));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#operator/=)
    ///
    /// ``` self: QtC.QSizeF, c: f64 ```
    pub fn OperatorDivideAssign(self: ?*anyopaque, c: f64) QtC.QSizeF {
        return qtc.QSizeF_OperatorDivideAssign(@ptrCast(self), @floatCast(c));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#toSize)
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn ToSize(self: ?*anyopaque) QtC.QSize {
        return qtc.QSizeF_ToSize(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qsizef.html#dtor.QSizeF)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QSizeF ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSizeF_Delete(@ptrCast(self));
    }
};
